### ISSUE TITLE
Add new operator for "equal as a set"

### ISSUE THREAD

**Issue by hyrodium** [JuliaGeometry/Meshes.jl#120]
(This issue is a continuation from #105.)

Currently, some `==` operator doesn't have transitivity. For example:

```julia
l1 = Line{2, Float64}(Point(0.0, 0.0), Point(1.0, 2.01027582027942e-9))
l2 = Line{2, Float64}(Point(0.0, 0.0), Point(1.0, 0.0))
l3 = Line{2, Float64}(Point(0.0, 0.0), Point(1.0, -1.4337470895339735e-8))

l1 == l2  # true
l2 == l3  # true
l1 == l3  # false
```

But, the julia language really care about the transitivity of `Base.==`, so we would like to deal with this problem!

---

What we have to do is the following:
1. Decide a new operator for "equal as a set" (maybe `\sim` or `\simeq`?).
2. List up all `==` methods for equallity as a set in the current repository.
3. Replace `==` with new operator, and add warning for deprecated `==`.
4. Replace deprecated `==` with other operation such as "equal as a map".

**Comment by juliohm** [comment]
Amazing plan @hyrodium. I would like to discuss which properties of these relations we are after.

In general we can talk about properties such as symmetry, transitivity, reflexivity in an [equivalence relation](https://en.wikipedia.org/wiki/Equivalence_relation). "equal as a set" seems to be just another equivalence relation on these objects. Julia has `\equiv` for equivalence relations for example.

Can you summarize the different types of relations you think will be useful to many users? After we have a list of such operations with their properties we will be able to reason more clearly about what to implement and what to deprecate.

**Comment by juliohm** [comment]
@hyrodium do you have any update regarding this issue? We should probably convert it into a Zulip thread otherwise. There are no clear action items because we still miss the properties of interest for these relations.

**Comment by hyrodium** [comment]
@juliohm Sorry for being late. I couldn't have enough time to deal with this issue. Let's discuss this on Zulip.

**Comment by juliohm** [comment]
Sure, we can polish these ideas further on Zulip and then if it becomes a pressing issue with clear action items we can reopen it here on GitHub and start working on a PR.
